src/box/memtx_tree.cc

```diff
@@ -346,12 +356,14 @@ static void
 tree_iterator_free(struct iterator *iterator)
 {
 	struct tree_iterator<USE_HINT> *it = get_tree_iterator<USE_HINT>(iterator);
-	tree_iterator_set_current<USE_HINT>(it, NULL);
+	if (it->last.tuple != NULL)
+		tuple_unref(it->last.tuple);
+	tree_iterator_set_last_hint(it, HINT_NONE);
```

Comment:
Please don't use `tree_iterator_set_last_hint` here - simply unref `last_func_key`, like you unref the last tuple.

Reply:
Done